### PR TITLE
[MCH] New definition of digit time and raw codec tests reorganisation

### PR DESCRIPTION
**Pull request by aphecetche**
The new definition combines the SAMPA time stamp, the SAMPA bunch
crossing counter, and the orbit number of the HBFrame containing the
digit.

The new definition does not change the size of the Digit structure.

The introduction of this new time definition requires adding the bunch
crossing counter to the SampaCluster structure, and therefore to change
the Elink decoders as well as some of the unit tests.

Talking about unit tests, the tests based on reference buffers have been
removed.  The creation of those buffers was a bit too fragile and quite
cumbersome. Replaced with the closure codec test instead.

### CODE REVIEW

**Comment by aphecetche** [comment]
@aferrero2707 will see if the checks are green, but this one is meant to replace #3618 (it includes all your modifications from 3618 plus a reorganisation of the raw codec tests).

**Comment by aphecetche** [comment]
@shahor02 this one is ready to be reviewed/merged

File: Detectors/MUON/MCH/Base/include/MCHBase/Digit.h [hunk 1]
```diff
@@ -28,14 +28,32 @@ namespace mch
...
+      struct {                       ///
+        uint32_t sampaTime : 10;     /// bit 0 to 9: sampa time
+        uint32_t bunchCrossing : 20; /// bit 10 to 29: bunch crossing counter
+        uint32_t reserved : 2;       /// bit 30 to 31: reserved
+        uint32_t orbit;              /// bit 32 to 63: orbit
+      };                             ///
+    };
+    uint64_t getBXTime()
+    {
+      uint64_t result = bunchCrossing;
```
Comment (shahor02) [inline Detectors/MUON/MCH/Base/include/MCHBase/Digit.h:44]: ``result`` is assigned and not used? I guess you wanted to have ``return result + (sampaTime * 4);``?

File: Detectors/MUON/MCH/Workflow/src/DataDecoderSpec.cxx [hunk 2]
```diff
@@ -132,17 +137,20 @@ class DataDecoderTask
       auto cruId = o2::raw::RDHUtils::getCRUID(rdhPtr);
       auto endpoint = o2::raw::RDHUtils::getEndPointID(rdhPtr);
       o2::raw::RDHUtils::setFEEID(rdhPtr, cruId * 2 + endpoint);
```
Comment (shahor02) [inline Detectors/MUON/MCH/Workflow/src/DataDecoderSpec.cxx:139]: This modifies the O2 message payload in the shared memory. Is this really needed ? From the code it is not obvious.

File: Detectors/MUON/MCH/Base/include/MCHBase/Digit.h (hunk 1 above)
Comment (aphecetche) [reply Detectors/MUON/MCH/Base/include/MCHBase/Digit.h]: You are correct. Fixed.

File: Detectors/MUON/MCH/Workflow/src/DataDecoderSpec.cxx (hunk 2 above)
Comment (aphecetche) [reply Detectors/MUON/MCH/Workflow/src/DataDecoderSpec.cxx:139]: I'll let @aferrero2707 confirm but this is currently needed as the raw data taken so far by mch (maybe not the latest latest with RDHv6) does not have the expected FEEID field correctly set. And the feeId value is central piece of our electronic mapping so it must be correct for the decoder to behave correctly.
This "patchPage" part will be removed eventually.

Comment (aferrero2707) [reply Detectors/MUON/MCH/Workflow/src/DataDecoderSpec.cxx:139]: @shahor02 I confirm what @aphecetche said: the patching of the RDH is only needed now as long as we cannot properly set the FEEID when configuring the CRU. We will probably keep a similar function in order to store in memory some of the RDH fields, but there will be no need to change the contents of the memory.

**Comment by shahor02** [comment]
ok, thanks, will merge once tests are done.

**Pull request merged**